### PR TITLE
GH-373: Add size/alignment checks to BindBufferMemory and BindImageMemory

### PR DESCRIPTION
**Pull request by cdwfs**
In addition to the new VU checks/tests for #373, this PR refactors Insert*MemoryRange() into pre/post-call format, and fixes some copy-paste errors in existing BindBufferMemory log_msg() calls.

### CODE REVIEW

**Comment by mark-lunarg** [comment]
Hey Cort, first character after the : in commit messages should be capitalized and the subject line should be right at 50 characters.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -5478,26 +5513,38 @@ static bool PreCallValidateBindBufferMemory(layer_data *dev_data, VkBuffer buffe
...
+                            buffer_handle, __LINE__, VALIDATION_ERROR_02174, "DS",
                             "vkBindBufferMemory(): memoryOffset is 0x%" PRIxLEAST64
                             " but must be an integer multiple of the "
                             "VkMemoryRequirements::alignment value 0x%" PRIxLEAST64
                             ", returned from a call to vkGetBufferMemoryRequirements with buffer. %s",
                             memoryOffset, buffer_state->requirements.alignment, validation_error_map[VALIDATION_ERROR_02174]);
         }
 
+        // Validate memory requirements size
+        if (buffer_state->requirements.size > mem_info->alloc_info.allocationSize - memoryOffset) {
```
Comment (cdwfs) [inline layers/core_validation.cpp]: Done

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -5539,6 +5586,12 @@ static void PostCallRecordBindBufferMemory(layer_data *dev_data, VkBuffer buffer
     std::unique_lock<std::mutex> lock(global_lock);
     auto buffer_state = GetBufferState(dev_data, buffer);
```
Comment (cdwfs) [inline layers/core_validation.cpp]: Done (in BindImageMemory as well)

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -5339,7 +5366,7 @@ static bool InsertMemoryRange(layer_data const *dev_data, uint64_t handle, DEVIC
         auto check_range = &obj_range_pair.second;
         bool intersection_error = false;
         if (rangesIntersect(dev_data, &range, check_range, &intersection_error)) {
-            skip_call |= intersection_error;
+            assert(!intersection_error);  // should have been caught in ValidateInsertMemoryRange()
```
Comment (cdwfs) [inline layers/core_validation.cpp]: Yeah, I should give this another look before I push.

Comment (cdwfs) [inline layers/core_validation.cpp]: I like the idea of letting callers opt out of the validation check in rangesIntersect(). I'm _almost_ tempted to hoist the whole check out into the caller, but there are too many call sites (and thus too much code duplication) for that . Would you prefer that I added a separate parameter to skip the check, or would it be sufficient to add an "if skip_call == NULL" check inside rangesIntersect(), and have calls from the state-update path pass NULL?

**Pull request merged**